File: backend-node/output.py
import firebase_admin
from firebase_admin import credentials, storage

# Initialize Firebase Admin SDK
cred = credentials.Certificate('../backend/serviceAccountKey.json')
firebase_admin.initialize_app(cred, {
    'storageBucket': 'mindmapwebsitepdf.appspot.com'  # Your Firebase Storage bucket name
})

# Function to get the content of a specified file in Firebase Storage
def get_file_content(file_path):
    bucket = storage.bucket()
    blob = bucket.blob(file_path)

    try:
        content = blob.download_as_text()  # Download the file content as text
        return content  # Return the content instead of printing it
    except Exception as error:
        return f'Error downloading file {file_path}: {error}'

# Example usage
if __name__ == '__main__':
    file_path = 'process/cleaned_summary.txt'  # Specify the file path in Firebase Storage
    content = get_file_content(file_path)
    print(content)  # This line can be removed if you don't want any output in console

File: backend-node/output.js
// Import Firebase Admin SDK
const admin = require('firebase-admin');
const fs = require('fs');
const path = require('path');

// Initialize Firebase Admin SDK
const serviceAccount = require(path.resolve(__dirname, '../backend/serviceAccountKey.json'));

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  storageBucket: 'mindmapwebsitepdf.appspot.com'  // Your Firebase Storage bucket name
});

// Function to print the content of a specified file in Firebase Storage
async function printFileContent(filePath) {
  const bucket = admin.storage().bucket();
  const file = bucket.file(filePath);

  try {
    const contents = await file.download(); // Download the file content as a Buffer
    console.log(`Content of ${filePath}:`);
    console.log(contents.toString('utf-8')); // Convert Buffer to string and print it
  } catch (error) {
    console.error(`Error downloading file ${filePath}:`, error);
  }
}

// Example usage
(async () => {
  const filePath = 'process/cleaned_summary.txt';  // Specify the file path in Firebase Storage
  await printFileContent(filePath);
})();
